src/libraries/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/PkcsHelpers.cs

```diff
@@ -575,8 +571,10 @@ public static DateTime DecodeUtcTime(byte[] encodedUtcTime)
 
         public static string DecodeOid(byte[] encodedOid)
         {
+            ReadOnlySpan<byte> invalidEmptyOid = new byte[] { 0x06, 0x00 };
+
             // Windows compat.
-            if (s_invalidEmptyOid.AsSpan().SequenceEqual(encodedOid))
+            if (invalidEmptyOid.SequenceEqual(encodedOid))
```

Comment:
Would it be better to just do:
```C#
if (encodedOid?.Length == 2 && encodedOid[0] == 6 && encodedOid[1] == 0)
```
?

Reply:
Also a good idea. I changed the method's parameter from `byte[]` to `ReadOnlySpan<byte>` to remove the need for the null check and switched to `AsnValueReader`. The only caller of this method is here 

https://github.com/dotnet/runtime/blob/2d1f7b027e34a552dc83e4d4082ca8320f3a476a/src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/Pkcs9ContentType.cs#L48-L51

And already did the null check, so the implicit conversion doesn't come in to play.

Reply:
I also considered making a similar change to `pSpecifiedDefaultParameters` but since it is a negative condition it seemed to make things less readable.